reader/receivers.py

```diff
@@ -84,9 +84,10 @@ def redirect_chapter(sender: Type[Chapter], instance: Chapter, **kwargs):
     new_dir = instance.get_directory()
     if old_dir != new_dir:
         if current.volume != instance.volume:
-            _move(old_dir.name, new_dir.name)
+            _move(str(old_dir.parent), str(new_dir.parent))
+            old_dir = new_dir.parent / old_dir.name
```

Comment:
It is. This functionality was completely broken before. `.name` returns just the name of the item and no other path information, so when that goes to `_move` it tries to do `settings.MEDIA_ROOT / .name` which obviously does not exist since the underlying context is missing. Plus the dirs never point to the parent to begin with they point to the bottom level chapter folder. What you need to do when the `volume` itself has changed is rename the volume folder as well, but once that happens old_dir is no longer comparable to new_dir. Which is why old_dir needs to be reassigned after the rename.

This would only not be necessariy if you could somehow rename recursively (which is not what _move does here at all) but even then `.name` is just wrong and if you could rename recursively why would you even need to stage the changes between `volume` and `chapter`?